### PR TITLE
gnrc_ipv6_nib: fix preparation of delayed NA

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
Currently the constructed NA for a delayed NA case is neither used nor released nor does it get an IPv6 header to be used properly. This fixes that case.

### Testing procedure
Since this is currently an edge-case we shouldn't really enter ATM, I fear you have to read and understand the code ... But the test case proposed by @aabadie proposed in https://github.com/RIOT-OS/Release-Specs/issues/98#issuecomment-461587132 might be one to test if together with #10978 this reduces (or even eliminates) the leaks in the packet buffer.

### Issues/PRs references
I'm still not sure this fixes all leaks (this code shouldn't even be entered since we do not use anycasts addresses at least in the test case these leaks were found), but definitely closes more leaks together with #10978.

### CODE REVIEW

**Comment by miri64** [comment]
Rebased to current master to ease testing.

**Comment by miri64** [comment]
@aabadie do you still have leaks with this PR?

**Comment by miri64** [comment]
If yes, please provide a dump of the packet buffer... I can't get any leaks anymore.

**Comment by aabadie** [comment]
@miri64, tested several times and no more leaks! Good job!

Codewise I can't tell if the changes are good or not. Maybe @cgundogan can tell ?

File: sys/net/gnrc/network_layer/ipv6/nib/nib.c [hunk 1]
```diff
@@ -796,20 +811,23 @@ static void _send_delayed_nbr_adv(const gnrc_netif_t *netif,
...
-        /* usually this should be the case, but when NCE is full, just
-         * ignore the sending. Other nodes in this anycast group are
-         * then preferred */
-        _evtimer_add(nce, GNRC_IPV6_NIB_SND_NA, &nce->snd_na,
-                     random_uint32_range(0, NDP_MAX_ANYCAST_MS_DELAY));
+    pkt = gnrc_ipv6_hdr_build(payload, NULL, &ipv6_hdr->src);
+    if ((payload = _check_release_pkt(pkt, payload)) == NULL) {
+        return;
+    }
+    ((ipv6_hdr_t *)payload->data)->hl = 255;
```
Comment (cgundogan) [inline sys/net/gnrc/network_layer/ipv6/nib/nib.c]: why `255` and not the configured default for the interface?

Comment (miri64) [reply sys/net/gnrc/network_layer/ipv6/nib/nib.c]: Because NAs [must have their hop-limit set to 255](https://tools.ietf.org/html/rfc4861#section-4.4) to ensure they are link-local. NAs with any other hop-limit value [are to be ignored silently](https://tools.ietf.org/html/rfc4861#section-7.12).

Comment (aabadie) [reply sys/net/gnrc/network_layer/ipv6/nib/nib.c]: Maybe this value could be provided as a define with some documentation ?

Comment (miri64) [reply sys/net/gnrc/network_layer/ipv6/nib/nib.c]: Sure, but since I use this magic number also in `gnrc_ndp_*_send()` I'd rather do this in a follow-up, so I don't need to touch these functions in a PR that needs to be backported.

Comment (cgundogan) [reply sys/net/gnrc/network_layer/ipv6/nib/nib.c]: >Sure, but since I use this magic number also in `gnrc_ndp_*_send()` I'd rather do this in a follow-up, so I don't need to touch these functions in a PR that needs to be backported.

:+1:

**Comment by miri64** [comment]
Backport provided in #11009

**Comment by miri64** [comment]
I think that concludes at least my part of release critical bugs that needed backporting. Let me know, if you need anything else for RC3.
